packages/core/src/api/composed/spore/transferSpore.ts

```diff
@@ -118,3 +118,124 @@ export async function transferSpore(props: {
+
+  // Inject live spore to Transaction.inputs and Transaction.outputs
+  for (const outPoint of props.outPoints) {
+    const sporeCell = await getSporeByOutPoint(outPoint, config);
+    const sporeScript = getSporeScript(config, 'Spore', sporeCell.cellOutput.type!);
+    const injectLiveSporeCellResult = await injectLiveSporeCell({
+      txSkeleton,
+      cell: sporeCell,
+      addOutput: true,
+      updateOutput(cell) {
+        cell.cellOutput.lock = props.toLock;
+        if (props.updateOutput instanceof Function) {
+          cell = props.updateOutput(cell);
+        }
+        return cell;
+      },
+      capacityMargin: props.capacityMargin,
+      defaultWitness: props.defaultWitness,
+      updateWitness: props.updateWitness,
+      since: props.since,
+      config,
+    });
+    txSkeleton = injectLiveSporeCellResult.txSkeleton;
+    injectLiveSporeCellResults.push(injectLiveSporeCellResult);
+  }
+
+  // Generate TransferSpore actions
+  const actions: (Partial<Pick<{ scriptInfoHash: string; scriptHash: string; data: string }, never>> &
+    Pick<{ scriptInfoHash: string; scriptHash: string; data: string }, 'data' | 'scriptInfoHash' | 'scriptHash'>)[] =
+    [];
+  for (const injectLiveSporeCellResult of injectLiveSporeCellResults) {
+    const actionResult = generateTransferSporeAction({
+      txSkeleton,
+      inputIndex: injectLiveSporeCellResult.inputIndex,
+      outputIndex: injectLiveSporeCellResult.outputIndex,
+    });
+    actions.push(...actionResult.actions);
+  }
+
+  if (!useCapacityMarginAsFee) {
+    // Inject needed capacity from fromInfos and pay fee
+    const injectCapacityAndPayFeeResult = await injectCapacityAndPayFee({
+      txSkeleton,
+      fromInfos: props.fromInfos!,
+      changeAddress: props.changeAddress,
+      feeRate: props.feeRate,
+      updateTxSkeletonAfterCollection(_txSkeleton) {
+        // Inject CobuildProof
+        const injectCobuildProofResult = injectCommonCobuildProof({
+          txSkeleton: _txSkeleton,
+          actions: actions,
+        });
+        _txSkeleton = injectCobuildProofResult.txSkeleton;
+
+        return _txSkeleton;
+      },
+      config,
+    });
+    txSkeleton = injectCapacityAndPayFeeResult.txSkeleton;
+  } else {
+    // Inject CobuildProof
+    const injectCobuildProofResult = injectCommonCobuildProof({
+      txSkeleton: txSkeleton,
+      actions: actions,
+    });
+    txSkeleton = injectCobuildProofResult.txSkeleton;
+
+    // choose an output to pay the fee
+    for (const item of injectLiveSporeCellResults) {
+      // Pay fee by the spore cell's capacity margin
+      try {
+        const tkSkeletonNew = await payFeeByOutput({
+          outputIndex: item.outputIndex,
+          txSkeleton,
+          config,
+        });
+        txSkeleton = tkSkeletonNew;
+        break;
+      } catch {
+        // if an error throwed, try next one, until it loops out
+        continue;
+      }
+    }
+  }
+
+  return txSkeleton;
```

Comment:
I recommend to follow its single version that returns transaction skeleton with transferred spore cells index in `inputs` and `outputs`

Reply:
others, LGTM

Reply:
added